src/metadata.cc

```diff
@@ -285,3 +300,21 @@ Napi::Value metadata(const Napi::CallbackInfo& info) {
 
   return info.Env().Undefined();
 }
+
+const char *PNG_COMMENT_START = "png-comment-";
+const int PNG_COMMENT_START_LEN = strlen(PNG_COMMENT_START);
+
+static void* readPNGComment(VipsImage *image, const char *field, GValue *value, void *p) {
+  MetadataComments *comments = reinterpret_cast<MetadataComments *>(p);
```

Comment:
Excellent use of `vips_image_map()` to read out the values here. Is is possible to switch to the slightly safer `static_cast`?

Reply:
Yes, static_cast works here, updated as suggested :+1: